pandas/tests/frame/methods/test_astype.py

```diff
@@ -587,3 +587,27 @@ def test_astype_ignores_errors_for_extension_dtypes(self, df, errors):
             msg = "(Cannot cast)|(could not convert)"
             with pytest.raises((ValueError, TypeError), match=msg):
                 df.astype(float, errors=errors)
+
+    def test_astype_tz_conversion(self):
```

Comment:
this test passes on master.  i dont see how it is related to read_json

Reply:
This PR originally started to fix only `read_json`. @mroeschke suggested a better fix was to correct `astype` conversion which would then cascade the fix to `read_json`.

The 'key' added test: `test_astype_tz_object_conversion(self, tz)` uses the specific, more basic, functionality from `test_astype_tz_conversion(self)`, so I included it for completeness (even though it already works). However, it may be superfluous, or there may be a similar test elsewehere in pandas that I could not find.

Let me know if this is OK to keep, or if I should remove this test.